src/frontend/initApp.js

```diff
@@ -9,19 +9,28 @@ import { RelayRoutingContext } from 'react-router-relay';
 import createStore from './createStore';
 import init from './init';
 import routes from './routes';
+import RelayNetworkLayer from './RelayNetworkLayer';
 
 import { ReduxIntlContainer } from 'frontend/bundles/common/components/ReduxIntl';
+import { addNotification } from 'frontend/bundles/notification/actionCreators';
 
 export default async function initApp() {
+  const store = createStore();
+
+  const onRelayError = (error) => {
+    store.dispatch(addNotification({
+      important: true,
+      text: error.message,
+    }));
+  };
+
   try {
     Relay.injectNetworkLayer(
-      new Relay.DefaultNetworkLayer('/graphql', {
+      new RelayNetworkLayer('/graphql', {
```

Comment:
I think here should be more flexible API. For example we could merge fetch options with other network layer options:
```js
new CustomNetworkLayer('/graphql', {
  fetchOptions: {
    credentials: 'same-origin',
  },
  onError,
});
```